### ISSUE TITLE
model creation

### ISSUE THREAD

**Issue by awegsche** [pylhc/omc3#292]
### add `Fetcher` classes / functions that link lattice definition (madx, sequence, strength files ...) to the model directory:
- `PathFetcher`: user specifiies path, this path is then linked to the model directory
- `AfsFetcher`: user specifies opitcs year (2018, 2022, etc) which is then linked to the model directory
- `LsaFetcher`: looks up info from LSA
### help commands to inform the user about possibilities and workflows
- list command args / usage
```
python omc3/model_creator.py \
    --accel lhc --type nominal --beam1 \
    --fetch afs --show_help
```
possible output:
```
AfsFetcher: usage
Links optics from afs. Available options:
run / year / tune definition from accelerator class
--optics_file        choose the optics
--list opticsfiles   shows a list of available opticsfiles for given settings
                     (run and year have to be set)
--show_help          show this help message and exit without creating a model
*** dry-run: no model created ***
```

- list possible modifers

```
python omc3/model_creator.py \
    --accel lhc --type nominal --beam1 \
    --fetch afs --list_modifiers
```
possible output:
```
AfsFetcher: list modifiers
opticsfile.1
opticsfile.30
opticsfile.30_ctpps1
*** dry-run: no model created ***
```

**Comment by JoschD** [comment]
Related to model creation:
- when to use `match_tunes` and `match_tunes_ats` macros (seems the former does not work for 2018+ optics, even non ATS?)
- when to use `coupling_knobs` and `coupling_knobs_ats`?  Are the definitions there still valid?

**Comment by mihofer** [comment]
> Related to model creation: How to handle the macros.
> Examples :
> 
> * when to use `match_tunes` and `match_tunes_ats` macros (seems the former does not work for 2018+ optics, even non ATS?)
> * when to use `coupling_knobs` and `coupling_knobs_ats`?  Are the definitions there still valid?

this is mostly/only relevant for LHC 
sometime between 2017 and 2018 (and earlier for some other optics) variable names were changed and broke some macros 
one could extend the macros with more if year/optics/md but that might become quite messy quite soon so rather we could move the lhc macros to the directories where the `seq` are stored and keep then keep macros names and interface the same (hopefully)

**Comment by awegsche** [comment]
`lhc_model_creator` looks for modifiers in the model dir:

https://github.com/pylhc/omc3/blob/3272f6f11e11bfe576a4c0615a69734c0064c591/omc3/model/model_creators/lhc_model_creator.py#L112-L117

is this the desired behaviour? Should then modifiers from e.g. acc-models be copied to the model dir?

**Comment by JoschD** [comment]
Yes, this is desired behaviour for relative paths. As written in the comment above the lines you were quoting, if you take absolute paths, they will stay absolute paths.
https://github.com/pylhc/omc3/blob/3272f6f11e11bfe576a4c0615a69734c0064c591/omc3/model/model_creators/lhc_model_creator.py#L111-L117

 So if you give it modifiers from acc-models you can give it absolute paths to the files.

**Comment by JoschD** [comment]
Implemented in #432
